cdap-watchdog/src/main/java/co/cask/cdap/metrics/process/MessagingMetricsProcessorService.java

```diff
@@ -122,14 +111,12 @@
                                    @Assisted Set<Integer> topicNumbers,
                                    @Assisted MetricsContext metricsContext,
                                    @Assisted Integer instanceId) {
-    this(cConf, datasetFramework, metricDatasetFactory, messagingService,
-         schemaGenerator, readerFactory, metricStore, topicNumbers, metricsContext, 1000, instanceId,
-         cConf.getBoolean(Constants.MetricsProcessor.METRICS_DATA_MIGRATION_SKIP, false));
+    this(cConf, metricDatasetFactory, messagingService,
+         schemaGenerator, readerFactory, metricStore, topicNumbers, metricsContext, 1000, instanceId);
```

Comment:
Is instanceId used anywhere other than for metrics migration? If not should we remove that too?

Reply:
The instance id is not required as a field in the class and I already removed it, it still needs to be passed in the constructor to construct several metric names, e.g, `processMetricName`, `metricsPrefixForDelayMetrics`